src/aws/auth.rs

```diff
@@ -1,16 +1,50 @@
 use std::time::Duration;
 
 use aws_config::{
-    default_provider::credentials::DefaultCredentialsChain, sts::AssumeRoleProviderBuilder,
+    default_provider::credentials::DefaultCredentialsChain, imds, sts::AssumeRoleProviderBuilder,
 };
 use aws_types::{credentials::SharedCredentialsProvider, region::Region, Credentials};
+use serde_with::serde_as;
 use vector_common::sensitive_string::SensitiveString;
 use vector_config::configurable_component;
 
 // matches default load timeout from the SDK as of 0.10.1, but lets us confidently document the
 // default rather than relying on the SDK default to not change
 const DEFAULT_LOAD_TIMEOUT: Duration = Duration::from_secs(5);
 
+/// IMDS Client Configuration for authenticating with AWS.
+#[serde_as]
+#[configurable_component]
+#[derive(Copy, Clone, Debug, Derivative)]
+#[derivative(Default)]
+#[serde(deny_unknown_fields)]
+pub struct ImdsAuthentication {
+    /// Number of IMDS retries for fetching tokens and metadata.
+    #[serde(default = "default_max_attempts")]
+    #[derivative(Default(value = "default_max_attempts()"))]
+    max_attempts: u32,
+
+    /// Connect timeout for IMDS.
+    #[serde(default = "default_timeout")]
+    #[serde_as(as = "serde_with::DurationSeconds<u64>")]
+    #[derivative(Default(value = "default_timeout()"))]
+    connect_timeout: Duration,
```

Comment:
```suggestion
    connect_timeout_seconds: Duration,
```

Reply:
We should use `#[serde(rename = "connect_timeout_seconds")]` to influence how's identified in a configuration file, but since it's a `Duration` on the Rust side, `connect_timeout` is the right name.

Reply:
🤔 @jszwedko what's your thought on this? Don't want to send @kevinpark1217 on a loop of renamings 😄

Reply:
What @tobz said sounds right to me. Internally `connect_timeout` seems fine to me. Externally it should be `connect_timeout_seconds`.